### DISCUSSION TITLE
How to read file contents in a custom slash command?

### DISCUSSION THREAD

**Discussion by jupcan** [zed-industries/zed#17714] [Feature Requests]
I've been attempting to create a custom slash command recently, but I'm unsure how to interact with the current working directory. The **Worktree** from **zed_extension_api** provides a method to obtain the root path, but I've encountered difficulties when trying to iterate over its files and read their content by combining it with _worktree.read_text_file(str)_.

My goal is to retrieve the content of all files in the directory, or at least those with specific extensions. This would allow me to call an external API with each file content. For a given test file relative path, it seems to work pretty well but can not iterate over all of them. I'm uncertain if this is currently feasible given the current state of Zed's extensions. Any guidance would be greatly appreciated!

``` rust
let worktree = worktree.ok_or("no worktree available")?;
let root = worktree.root_path();
let mut file_contents = String::new();
file_contents.push_str(&format!("root path: {}\n\n", root));

let specific_file = "src/test_file.rs";
match worktree.read_text_file(specific_file) {
    Ok(content) => {
        file_contents.push_str(&format!("content of {}:\n\n{}\n\n", specific_file, content));
    },
    Err(e) => {
        file_contents.push_str(&format!("error reading {}: {}\n\n", specific_file, e));
    }
}

Ok(SlashCommandOutput {
    sections: vec![SlashCommandOutputSection {
        range: (0..file_contents.len()).into(),
        label: "File Contents Descriptions".to_string(),
    }],
    text: file_contents,
})
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
